src/cpsr.rs

```diff
@@ -15,7 +15,7 @@ impl Cpsr {
     }
 
     fn signed(&self) -> bool {
-        self.0 & 0x8000 != 0
+        self.0 & 0b1000_0000_0000_0000 != 0
```

Comment:
I have only converted to binary `0x80_00` , but it shouldn't be `0x80_00_00_00`?

`sign_flag` is 31st bit ([Documentation](https://rust-console.github.io/gbatek-gbaonly/#armcpuflagsconditionfieldcond))
![Immagine 2022-09-09 002431](https://user-images.githubusercontent.com/99926684/189236604-3a57ee4e-91ae-4e2f-840e-829ab9109f9b.png)

Reply:
Can you add a test that should verify this error? If this is true, open a new PR because is important

Reply:
I'll try, I'm still learning rust, I've been studying it for 5 days.

Reply:
I opened a new PR.
Do you want to keep this open to fix the error? (From 0xAAAA to 0xAAAAAAAA)?